src/pip/_internal/cli/cmdoptions.py

```diff
@@ -568,6 +568,25 @@ def prefer_binary():
          'if this option is used.'
 )  # type: Any
 
+use_pep517 = partial(
+    Option,
+    '--use-pep517',
```

Comment:
Instead of a catch-all, we should switch this to be something that uses FormatControl-like class to enable/disable support per-package name or for `:all:`.

Reply:
There's even some merit in doing that for the already-released `--no-build-isolation` perhaps; in a backwards compatible way, if we can figure one out for it.

Reply:
Again, a follow-up PR sounds the way to go here. But I view `--use-pep517` as very much a temporary measure anyway, in a couple of releases I expect it to vanish and `use_pep517=True` to be the only supported behaviour. So maybe just living with it makes sense too.